api/src/org/apache/cloudstack/api/command/admin/template/PrepareTemplateCmd.java

```diff
@@ -60,6 +61,15 @@
                description = "template ID of the template to be prepared in primary storage(s).")
     private Long templateId;
 
+    @ACL(accessType = AccessType.OperateEntry)
+    @Parameter(name = ApiConstants.STORAGE_ID,
+            type = CommandType.UUID,
+            entityType = StoragePoolResponse.class,
+            required = false,
+            description = "storage pool ID of the primary storage pool to which the template should be prepared. If it is not provided the template" +
+                    " is prepared on all the available primary storage pools.")
+    private Long storageId;
```

Comment:
@kishankavala There isn't a poolid or storagePoolid constant available in apiconstants. I had looked for it and couldn't find it. Moreover, storageId is the parameter name used through the apis whenever a primary storage pool is referred, for example in list/update/migrate volumes, volume response etc. So I decided to use storageId here too for consistency, instead of defining a new one.